app/uk/gov/hmrc/apinotificationpull/connectors/EnhancedApiNotificationQueueConnector.scala

```diff
@@ -25,6 +25,7 @@ import uk.gov.hmrc.apinotificationpull.model.{Notification, NotificationStatus,
 import uk.gov.hmrc.http.{HeaderCarrier, HttpResponse, NotFoundException, _}
 import uk.gov.hmrc.play.bootstrap.config.ServicesConfig
 import uk.gov.hmrc.http.HttpClient
+import uk.gov.hmrc.http.HttpReads.Implicits._
```

Comment:
Search everywhere in codebase for NotFoundException and replace with Upstream error response